native/src/modules/endpoint/reducers/pushEvent.js

```diff
@@ -71,22 +76,25 @@ const pushEvent = (state: CityContentStateType, action: PushEventActionType): Ci
     ? { ...state.resourceCache.value, ...resourceCache }
     : resourceCache
 
-  const newEventsRouteMapping = { ...state.eventsRouteMapping }
+  const newRouteMapping = { ...state.routeMapping }
 
   if (refresh) {
-    entries<EventRouteStateType>(state.eventsRouteMapping)
-      .filter(([_, route]) => city === route.city && path !== route.path && language === route.language)
+    entries<RouteStateType>(state.routeMapping)
+      .filter(([_, route]) => route.routeType === 'event' &&
+        city === route.city &&
+        path !== route.path &&
+        language === route.language)
       .forEach(([key, route]) => {
-        newEventsRouteMapping[key] = route.path
+        newRouteMapping[key] = route.path
```

Comment:
Why is the check not necessary here while it is for categories?

Reply:
I think it is related to the typing of the chained entries -> filter -> foreach. Flow sees the route stilla as a RouteStateType, as we cannot implicitly change the type after the filter call. But as the assignment in `pushEvent` allows the path to be undefined there is no check necessary. for the categories however the path has to be defined.